### ISSUE TITLE
jsdoc/check-tag-names doesn't support Web Component tags

### ISSUE THREAD

**Issue by anselmbradford** [gajus/eslint-plugin-jsdoc#1399]
There are a number of jsdoc tags that are used within web component definitions (see https://www.npmjs.com/package/web-component-analyzer#overview-of-supported-jsdoc-tags), such as `@element` and `@slot`. 

jsdoc/check-tag-names doesn't support these tags (see https://github.com/gajus/eslint-plugin-jsdoc/blob/main/.README/rules/check-tag-names.md) because they don't appear in the [JSDoc 3 Block Tags](https://jsdoc.app/#block-tags). 

What is the recommended way to handle these? Should I ignore them in the config?

**Comment by brettz9** [comment]
As per the `check-tag-names`, rule, you have the option to add the tag names you wish to allow to the `definedTags` option.

You can also set them via the `structuredTags` setting.

The latter has the advantage of working with other rules (in particular, `check-types`, `no-undefined-types`, and `valid-types`), such as if you wanted to define `@attr` like follows:

```js
      settings: {
        jsdoc: {
          structuredTags: {
            attr: {
              name: 'namepath-referencing',
              type: true,
              required: ['name', 'type']names
            }
```

Taking the following example:

```js
/**
 * @attr {on|off} switch
 */
```

...the config above will require that `switch` and `on` and/or `off` are added. Since it is not `namepath-defining`, one will not be able to use the word "switch" in other tags to point to it (e.g., by `@type` ).

Closing, as that should resolve, but feel free to comment further as needed.
